File: leetcode-solutions/ruby/387. First Unique Character in a String.rb
# https://leetcode.com/problems/first-unique-character-in-a-string/

# @param {String} s
# @return {Integer}
def first_uniq_char(s)
  map = {}
  s.each_char.with_index do |c,i|
      if map[c]
          map[c] += 1
      else
          map[c] = 1
      end
  end
  
  s.each_char.with_index do |c, i|
      return i if map[c] == 1
  end
  
  -1
end

File: leetcode-solutions/python/387. First Unique Character in a String.py
# https://leetcode.com/problems/first-unique-character-in-a-string/

class Solution:
    def firstUniqChar(self, s: str) -> int:
        memo = {}
        for i, c in enumerate(s):
            if c in memo:
                memo[c] += 1
            else:
                memo[c] = 1
        for i, c in enumerate(s):
            if memo[c] == 1:
                return i
        return -1
